artiq/tools.py

```diff
@@ -240,7 +240,16 @@ def atexit_register_coroutine(coroutine, loop=None):
 
 class TaskObject:
     def start(self):
-        self.task = asyncio.ensure_future(self._do())
+        async def log_exceptions(awaitable):
+            try:
+                return await awaitable()
+            except asyncio.CancelledError:
+                raise
+            except Exception:
+                logger.exception("Unhandled exception in TaskObject task body", exc_info=True)
```

Comment:
logger.exception?

Reply:
== `logger.error("error message", exc_info=True)`

Reply:
As I pointed before the rest of ARTIQ uses logger.error, so this should be consistent.